### DISCUSSION TITLE
Wireframe Prep Work

### DISCUSSION THREAD

**Discussion by donschaefer** [CareTogether/CareTogetherCMS#942] [Ideas]
**Convert 5/29 meeting notes (below) into something more organized/actionable for a designer to start creating wireframes from.**

**Things already on the Family Screen:**

- Family Documents & ability to upload/view/delete
- Family member names & basic demographic info (adults & children)
- For each adult:
  - Contact info
  - Approvals info (volunteer roles/applications/etc.)
  - User login/account/role info
- For each person:
  - Notes
  - Concerns
- Notes (timeline)
  - Notes can be tied to actions (requirements changes/completions or child location changes) & documents, but also other events that require notes to be added
  - Notes have an author & approver info as well as timestamps for creation/approval
- Toolbar of sorts for adding adults, children, notes; a delete family option
- Role removal
- Family level & individual level role approval status
- Family level & individual level approval requirements (incomplete, completed, exempted, or available applications)
  - Hiding available applications until they're desired would be useful
- Primary contact (could be merged into the adult "card"/table row/etc.)
- Custom fields @ family level
- "Home Church" field may become standard rather than custom
- Communities
- Referrals
  - Referral comments
  - Incomplete/Complete/Exempted requirements
  - Open/closed status & date
  - Ability to open & close
  - Ability to swap between referrals
  - Referral-level custom fields
- Arrangements
  - Filter
  - Add arrangements
  - Arrangement cards
    - Stage
    - ... whole bunch of stuff 🙂

**Things to add to the Family Screen:**
- Calendar view of significant upcoming & past dates for the family (approval requirements, child location changes, arrangement stuff, ...)
- Task view of significant upcoming & past-due items (to avoid having to skim everything)
- Search & filtering options for the Notes in the timeline
- At-a-glance view of where children are, rather than separated by arrangement
- People in the family "above the fold"
- [Longer term?] Split "referral" out from "case"
  - Tracking referral-level requirements
  - Tracking case-level requirements
  - Referral history separate from case history, arrangements live under the 'case'
- Family/personal goals - these will be tied to a "case" and to a person in the family.
  - Ability to break goals into individual smaller steps (SMART goal setting & progress tracking)
- Show which arrangements a volunteer family/member is/was assigned to
- Quality/progress survey tracking capability -- issue a follow-up survey to parents being served, e.g. at beginning & end of hosting and every six months, and track progress (quantitative metrics)
- Alerts area (?) for the family @ top of screen for any critical family-specific items
- Family/person "tags" - e.g., 'Spanish'
- Initiating DocuSign from a requirement
- Forms received for the family (Form.io)
- Family profile -- e.g., volunteers' ability/preferences to host ages/genders of kids, family photos
- Ability to text/email the family directly from this screen, possibly with message templates
- Integrate external systems' data:
  - Mailing lists
  - Donation records/profiles
  - Emails sent to staff user's account (?)

**Comment by donschaefer** [comment]
@LarsKemmann - below are some high-level thoughts/notes I'd like to review with you (possibly during our upcoming call) in relation to how we might reorganize existing screens a bit to better integrate the functionality you're looking to add...

**Re-purposed Dashboard:**
* Coming up this week (tasks coming due, upcoming events, etc)
* Goal Progress (family-specific)
* Quality/Progress Trend Graph (admin-specific / shows trend patterns of active families)
* Contacts (either by frequency of use, association to upcoming tasks/events, or some combination)

**Monthly Calendar (currently the function that the "Dashboard" serves)**
- How does this differ from "Calendar view of significant upcoming & past dates for the family (approval requirements, child location changes, arrangement stuff, ...)" except insofar as one is specific to a family? ... 
* could we achieve something similar with role-based filter defaults? 
* Could we drive the view based on query string & then link to specific calendar views from the various family pages?

**Rename "Volunteers" to "Families" & combine with existing "Referrals"**
* Display referrals in a column next to roles in a manner similar to how roles are displayed, but with the key difference that: 
* A given family member might have 1 expandable "chip" for "Pending Referrals (3)" & another for "Active Cases (2)"
* Clicking on an expandable chip would provide a bullet-point summary of the associated items (thus expanding the row in the process)
* Clicking on a row takes you to the individual "Family" screen


**Individual Family Screen:**
* Family Photo
* Members (add btn)
	> photo
	> Name, Age, Location, contact info, tags
		* Clicking email/sms # opens contact form to send email/sms (on mobile, could also offer the ability to call on sms #)
* Communities
* Volunteer preferences (should this go at the family level or should it be member-specific?)
* Referrals (bullet-point summary, clicking takes you separate Referral / Case screen)

**Referral / Case screen** (new / essentially a re-organized "Family" screen that is more "referral"-focused)
	> Alert area (situational)
	> Comments (includes edit btn)
	> Stats (includes edit btn)
	> Documents (includes add btn)
		* Integrate docusign
	> Goals (add, edit/delete, search btns)
		* Are goals specifically related to associated referral/case, or are some personal? 
		* What parallels can we draw between "Goals" & epics/features/user stories in devOps?
		* Each goal likely needs to be broken down into multiple tasks
	> Tasks (sort by deadline/importance?)
	> Notes (add, edit/delete, search controls)
	> Arrangements

**Comment by LarsKemmann** [comment]
@Rabi94 @PabloDinella Here is the list of desired future features to incorporate into UI design.
